Pool/Authenticate.vb

```diff
@@ -0,0 +1,275 @@
+﻿Imports System.Security.Cryptography
+Imports System.Text
+Imports Pool
+Imports Pool.Games
+Public Class Authenticate : Implements ILogin
+
+#Region "Props/Constr"
+    Private Structure HashFormat
+        Public length As Integer
+        Public salt As Byte()
+        Public hash As Byte()
+    End Structure
+
+    Public Enum PermissioninApp
+        Debug = -1
+        Admin = 0
+        User = 1
+        Unregister = 2
+    End Enum
+
+    Public Property User As String
+    Public Property Password As String
+    Public Property isLoggedIn As Boolean
+    Public Sub New()
+    End Sub
+    Public Sub New(user As String, password As String, isloggedIn As Boolean)
+        Me.User = user
+        Me.Password = password
+        Me.isLoggedIn = isloggedIn
+    End Sub
+    Public Sub New(user As String, password As String)
+        Me.User = user
+        Me.Password = password
+    End Sub
+#End Region
+
+    ''' <summary>
+    ''' Get hashed password for that user
+    ''' Generate hash from salt and lenght from DB
+    ''' Compare and return login status
+    ''' </summary>
+    Public Function GetLogin() As Boolean Implements ILogin.GetLogin
+        Dim isLoginCreds As Boolean = False
+        Dim hashedPasswordDS As New Games
+
+        'Get hashed string from db
+        Dim pwdHashDS = hashedPasswordDS.GetAllResults($"exec [selUserPwdHashed] @user='{User}'")
+        Dim hashpwdfromDB As String = pwdHashDS.Tables(0).Rows(0).Item("password")
```

Comment:
This will break if any non-registered user attempts to sign in. This should be updated to cover the case where we return no results from selUserPwdHashed.

Reply:
Returns an alert box for incorrect creds instead of crashing.